### DISCUSSION TITLE
Understanding _id of saved row

### DISCUSSION THREAD

**Discussion by spmvoss** [Budibase/budibase#10691] [Help] [answered]
I have a form that creates a part and a part variant in a single form. First, the part is created and I want to use the row id in the creation of the part variant. However, this does not seem to work as the {{ Action 1.Saved row._id }} is a string containing a URL-encoded array (for example, if the _id is 2, it returns %5B2%5D. 

I am probably missing something here. Am I supposed to parse this before using it in the second action? Why does it return an URL-encoded array in the first place? 

I am using Postgresql as the database. 

![image](https://github.com/Budibase/budibase/assets/4139244/c3f253a7-78b5-41ae-951b-aabe1e2dda87)
![image](https://github.com/Budibase/budibase/assets/4139244/9d1e1ef6-e18c-489e-8f9b-4d878007bc4c)

**Comment by melohagan** [comment, marked as answer]
Hey @spmvoss 

The **_id** is specific to Budibase. If you want to access the primary key of the saved row, can you use the actual field name, e.g. `{{ Action 1.Saved row.name_of_primary_key_field }}`

If not, you may need to use the **Fetch row** action, which will pull back the JSON row detail for a given row **_id**.

**Reply by spmvoss** [reply]
Thanks @melohagan! That makes a lot of sense actually, thank you, in that case indeed it is very simple and I can just use my primary key field :)
